scalecube-organization/src/main/java/io/scalecube/organization/operation/AddOrganizationApiKey.java

```diff
@@ -55,9 +60,17 @@ protected GetOrganizationResponse process(
       }
     }
 
-    ApiKey apiKey = ApiKeyBuilder.build(keyStore, organization, request);
+    String keyId = UUID.randomUUID().toString();
 
-    organization.addApiKey(apiKey);
+    try {
+      KeyPair keyPair = generateKeyPair(keyId);
+      ApiKey apiKey = ApiKeyBuilder.build(keyPair.getPrivate(), organization.id(), keyId, request);
+      organization.addApiKey(apiKey);
+    } catch (Exception ex) {
+      // failed to persist organization secret rollback
+      context.repository().deleteById(organization.id());
```

Comment:
Do we need to delete organization in case of exception during building ApiKey?

Reply:
Not need. Done.